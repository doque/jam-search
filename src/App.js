import React, { Component } from "react";
import { dataFetcher } from "./Api";

import Suggestions from "./Suggestions";
import SearchBar from "./SearchBar";

class App extends Component {
  state = {
    data: {
      suggests: []
    },
    selected: ""
  };

  clickHandler(selected) {
    this.setState({ selected });
  }

  changeHandler(event) {
    const value = event.target.value.trim();
    const type = event.target.name;

    if (value.length >= 3) {
      this.fetchData(value, type).then(data => this.setState({ data }));
    } else {
      this.setState({ data: { suggests: [] } });
    }
    this.setState({
      selected: value
    });
  }

  fetchData(value, type = "what") {
    const url = `https://jameda.localtunnel.me/suche.jameda-elements.de/${type}-new?query=${value}`;

    return dataFetcher(url);
  }

  render() {
    return (
      <div className="App">
        <SearchBar
          selected={this.state.selected}
          changeHandler={this.changeHandler.bind(this)}
        />
        <Suggestions
          data={this.state.data}
          clickHandler={this.clickHandler.bind(this)}
        />
      </div>
    );
  }
}

export default App;
