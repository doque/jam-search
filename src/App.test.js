import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

it('renders without crashing', () => {
  ReactDOM.render(<App />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it.only('fetches data from the server', (done) => {
  const myMock = jest.fn();
  App.prototype.fetchData = myMock;
  const div = document.createElement('div');  
  expect(App.prototype.fetchData.called).toBe(true);
});
