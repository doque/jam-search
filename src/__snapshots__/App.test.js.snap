// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <label>
      Was?
      <input
        name="what"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label>
      Wo?
      <input
        name="where"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <div />
</div>
`;
